### PR TITLE
Migrate to Pytest

### PR DESCRIPTION
**Pull request by vmuriart**
### What does this implement/fix? Explain your changes.

- `Unittests` module requires alot of boilerplate setup to get running. `Pytest` the process of unittest writing within python making it easier for new contributors to add tests.
- Any test can be run by itself again, instead of having to run the entire test suite.

### Does this close any currently open issues?

`N/A`

### Any other comments?

This was the original meant to be part of #329 but decided to separate it out.

### Checklist

Check all those that are applicable and complete.

-   [x] Make sure to include one or more tests for your change
-   [x] If an enhancement PR, please create docs and at best an example
-   [x] Add yourself to [`AUTHORS`](../blob/master/AUTHORS.md)
-   [x] Updated the [`CHANGELOG`](../blob/master/CHANGELOG.md)

### CODE REVIEW

**Comment by vmuriart** [comment]
Also another nice feature is that it has a nicer output

```
============================= test session starts ==============================
platform linux -- Python 3.3.5, pytest-3.0.6, py-1.4.32, pluggy-0.4.0
rootdir: /home/travis/build/pythonnet/pythonnet, inifile: setup.cfg
collected 360 items 
src/tests/test_array.py ........................................
src/tests/test_callback.py ..
src/tests/test_class.py ..................
src/tests/test_compat.py ................
src/tests/test_constructors.py ....
src/tests/test_conversion.py ....................
src/tests/test_delegate.py .................
src/tests/test_docstring.py ...
src/tests/test_engine.py .ss
src/tests/test_enum.py ...............
src/tests/test_event.py ............................
src/tests/test_exceptions.py .........................
src/tests/test_field.py ..............................
src/tests/test_generic.py .................
src/tests/test_import.py .
src/tests/test_indexer.py .............................
src/tests/test_interface.py ....
src/tests/test_method.py ........................................
src/tests/test_module.py ...........................
src/tests/test_property.py ..........
src/tests/test_recursive_types.py .
src/tests/test_subclass.py sssss.
src/tests/test_sysargv.py .
src/tests/test_thread.py ...
```

File: MANIFEST.in [hunk 1]
```diff
@@ -0,0 +1,20 @@
+graft docs
```
Comment (filmor) [inline MANIFEST.in]: What is this file for?

**Comment by denfromufa** [comment]
I had the same question about manifest.in

File: MANIFEST.in (hunk 1 above)
Comment (vmuriart) [inline MANIFEST.in]: I forgot to remove that commit/file. Its supposed to replace the `_get_source_files` in `setup.py`.
I didn't get around to testing it though. Removing it fixes the conda recipe. 

Let me fix that.

Comment (denfromufa) [inline MANIFEST.in]: Same question

**Comment by vmuriart** [comment]
@filmor @denfromufa conda recipe fixed.
I wrote that `MANIFEST.in` a while back, it was referring to a `docs` folder we had i think.

**Comment by vmuriart** [comment]
All builds finished, none of them had the conda.recipe build error from the manifest. Looks like we are good to merge then?

**Comment by denfromufa** [comment]
I like the switch to purest, can you give 1-2 days to review this and test locally?

**Comment by vmuriart** [comment]
Sure. Test wise, nothing changed other than indent level.
Have you tried using `tox` for testing? It will let you run against all interpreters you have installed.

**Comment by denfromufa** [comment]
Even PyPy ;)

**Comment by vmuriart** [comment]
You can configure it for it; though idk if pythonnet would have any issues w it.

**Comment by vmuriart** [comment]
@denfromufa did you get a chance to finish testing?

**Comment by denfromufa** [comment]
Sorry, PyPy got me distracted. Let me try pytest tonight. If not, then you
can merge tomorrow.

**Pull request merged**